src/future.ml

```diff
@@ -408,17 +515,29 @@ module Scheduler = struct
     let jobs =
       Hashtbl.fold running ~init:[] ~f:(fun ~key:_ ~data:job acc -> job :: acc)
     in
-    match jobs with
-    | [] -> ()
-    | first :: others ->
-      Format.eprintf "\nWaiting for the following jobs to finish: %t@."
-        (fun ppf ->
-           Format.fprintf ppf "[@{<id>%d@}]" first.id;
-           List.iter others ~f:(fun job ->
-             Format.fprintf ppf ", [@{<id>%d@}]" job.id));
-      List.iter jobs ~f:(fun job ->
-        let _, status = Unix.waitpid [] job.pid in
-        process_done job status ~exiting:true)
+    let rec wait_for_jobs msg_time jobs = match jobs with
+      | [] -> ()
+      | job :: jobs when msg_time > 0. ->
+         let pid, status = Unix.waitpid [WNOHANG] job.pid in
+         if pid <> 0 then begin
+           process_done job status ~exiting:true;
+           wait_for_jobs msg_time jobs
+         end else begin
+           let dt = 0.05 in
+           let _ = Unix.select [] [] [] dt in
+           wait_for_jobs (msg_time -. dt) (job :: jobs)
+         end
+      | jobs ->
+         let pp_job ppf job =
+           let (_, name, _) = split_prog job.job.prog in
+           Format.fprintf ppf "%s [@{<id>%d@}]" name job.id in
+         Format.eprintf "\nWaiting for the following jobs to finish: %a@."
```

Comment:
This is better, but the ids appear nowhere in non-verbose mode, which is a bit confusing. I think we should just print the number of background jobs without `--verbose`, or better print the list of jobs as they are printed by `process_done`. BTW I used these stanza to test this code:

```scheme
;; execute this to check the behavior when background jobs take time to finish:
;;
;; $ ./_build/default/bin/main.exe build @test/fail-with-background-jobs-running
;;
(alias
 ((name   sleep5)
  (action (system "sleep 5"))))

(alias
 ((name   sleep1-and-fail)
  (action (system "sleep 1; exit 1"))))

(alias
 ((name fail-with-background-jobs-running)
  (deps ((alias sleep5)
         (alias sleep1-and-fail)))))
```

This can go in `test/jbuild`